PWGCF/EbyEFluctuations/Tasks/RobustFluctuationObservables.cxx

```diff
@@ -1135,11 +1126,125 @@ struct RobustFluctuationObservables {
     }
 
     // ### event selection cuts
-    if (!collision.sel8())
+    if (!collision.sel8()) {
+      prevBC = collBC;
+      prevBcInTF = bcInTF;
+      prevOrbit = orbitInTF;
+      prevTF = TFid;
+      prevGlobalFoundBC = globalFoundBC;
+      // double prev_vZ_keepForBelow = prev_vZ;
+      prev_vZ = vZ;
+      prev_mult = nTracksAll;
+      prev_globTrkContrib = nTracksGlobalPVAccepted;
+
       return;
+    }
+
+    // ##### check how often we analyze collision in the same BC (and also the vZ difference)
+    if (collision.selection_bit(o2::aod::evsel::kNoTimeFrameBorder)) {
+      // auto prevColl = collision - 1;
+
+      if (prevBC != 9999) //&& orbitInTF >= prevOrbit )
+      {
+        int32_t diff = (int32_t)collBC - (int32_t)prevBC;
+        histosEvent.fill(HIST("hBC_DIFF_to_previous"), diff);
+        if (diff == 0)
+          histosEvent.fill(HIST("hBC_DIFF_to_previous_vZvZ_2D"), vZ, prev_vZ);
+      }
+
+      // global found BC:
+      if (prevGlobalFoundBC != 9999) // 0) //&& orbitInTF >= prevOrbit )
+      {
+        int32_t diff = (int32_t)globalFoundBC - (int32_t)prevGlobalFoundBC;
+        histosEvent.fill(HIST("hBC_DIFF_to_previous_FOUND_BC"), diff);
+
+        if (counterPVcontributorsAfterTPCcuts > 0) {
+          histosEvent.fill(HIST("hBC_DIFF_to_previous_FOUND_BC_1goodVertContribTPC"), diff);
+          if (isFT0 && cutVzTrackT0diffLower < diff_PV_ft0_tracks && diff_PV_ft0_tracks < cutVzTrackT0diffUpper) // cm
+            histosEvent.fill(HIST("hBC_DIFF_to_previous_FOUND_BC_1goodCont_FTvertexCut"), diff);
+        }
+
+        if (diff == 0) // QA vZ with prev_vZ correlations
+        {
+          histosEvent.fill(HIST("hBC_DIFF_to_previous_FOUND_BC_vZvZ_2D_diff0"), vZ, prev_vZ);
+          if (counterPVcontributorsAfterTPCcuts > 0) {
+            histosEvent.fill(HIST("hBC_DIFF_to_previous_FOUND_BC_1goodVertContribTPC_vZvZ_2D_diff0"), vZ, prev_vZ);
+            if (isFT0 && cutVzTrackT0diffLower < diff_PV_ft0_tracks && diff_PV_ft0_tracks < cutVzTrackT0diffUpper) // cm
+              histosEvent.fill(HIST("hBC_DIFF_to_previous_FOUND_BC_1goodCont_FTvertexCut_vZvZ_2D_diff0"), vZ, prev_vZ);
+
+            histosEvent.fill(HIST("hBC_DIFF_to_previous_FOUND_BC_mult_mult_2D_diff0"), nTracksAll, prev_mult);
+            histosEvent.fill(HIST("hBC_DIFF_to_previous_FOUND_BC_globTrkContrib_globTrkContrib_2D_diff0"), nTracksGlobalPVAccepted, prev_globTrkContrib);
+          }
+
+          fillHistForThisCut("NoTFborder_FoundBCwithDiff0", multNTracksPV, multTrk, nTracksGlobalAccepted, multT0A, multT0C, multV0A, t0cCentr);
+        }
+      }
```

Comment:
For readability I would suggest a consistent use of the curled brackets
My choice is
- shape
```
if (condition) {
  ...
} else {
  ...
}
```
- always use them, even for a single conditional line

But the important thing is being consistent in their use